rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -448,33 +447,17 @@ class NodeHandle_ : public NodeHandleBase_
   bool advertiseService(ServiceServer<MReq, MRes, ObjT>& srv)
   {
     bool v = advertise(srv.pub);
-    for (int i = 0; i < MAX_SUBSCRIBERS; i++)
-    {
-      if (subscribers[i] == 0) // empty slot
-      {
-        subscribers[i] = static_cast<Subscriber_*>(&srv);
-        srv.id_ = i + 100;
```

Comment:
What is the impact of losing this logic?

Reply:
It's exactly the same code as in `subscribe()`. Also, the return value has not changed; `advertiseService()` returns only `true` if both publisher and subscriber has been created; furthermore the subscriber is created regardless of the value of `v`.

Because `advertiseService` and `serviceClient` now only consist of two function calls (and the `&&` if we keep the return value), these functions might be more likely inlined by the compiler.

Reply:
Okay, yes, this makes sense. Thanks for explaining.